### ISSUE TITLE
[Nuxt] Cookiebot does not replace scripts when navigating with NuxtLink

### ISSUE THREAD

**Issue by wavedeck** [ambitiondev/cookiebot#5]
I use cookiebot with in manual blocking mode with "data-src" and "data-cookieconsent" attributes to block a Google Maps iframe and ensure no data is sent, as data protection laws are incredibly strict in my country.

When visiting the page directly or reloading it, the cookie consent works and loads the iframe when consent was given, but when using NuxtLink to navigate to the page, it will not load the iframe when consent was given previously.

Minimal reproduction example:

1. Follow the installation instructions
(I put the consentBanner into `layouts/default.vue` instead rather than `app.vue`)
2. Add a "iframe.vue" page and put the following:

```vue
<template>
    <section id="google-maps">
        <div class="container mx-auto px-8">
            <div class="cookieconsent-optout-marketing">
                <div class="content">
                    <span>To load Google Maps, you need to consent to our marketing settings</span>
                    <button class="bregtec-button" onclick="Cookiebot.renew()">Open settings</button>
                </div>
            </div>
            <iframe
                data-src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d16457.456161886992!2d16.373624362050386!3d48.21420021089018!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x476d0799bf2006df%3A0x8685168014c4425f!2sHofburg!5e0!3m2!1sde!2sat!4v1708284345348!5m2!1sde!2sat"
                data-cookieconsent="marketing" width="100%" height="100%" style="border: 0;" loading="lazy"
                referrerpolicy="no-referrer-when-downgrade"></iframe>
        </div>
    </section>
</template>
```

3. Add another page and put the following:

```vue
<template>
    <NuxtLink to="/iframe">
</template>
```
4. Open the second page in your browser and accept the consent.
5. Navigate to the iframe page by clicking on the NuxtLink.

Cookiebot will not load the iframe but since you already consented, you wont see anything.

**Comment by wavedeck** [comment]
I believe this is happening because cookiebot will not re-initialize when navigating using NuxtLink, as Nuxt will only replace the page contents and not perform a full page refresh (by design). This is potentially dangerous when relying on auto blocking mode and can lead to cookies being loaded, although they should not.

In my case (manual blocking mode) it's just preventing the iframe to work. Which is still not great but at least not a compliance risk.

**Comment by wavedeck** [comment]
I came up with a workaround, by using the Cookiebot SDK which is available on the window object when Cookiebot is loaded. Just run `Cookiebot.runScripts()` when the page mounts on every page that have scripts that require consent.

This approach works fine with NuxtLink, as this runs the script parser again when you navigate to the page.
The only drawback is, that it's a little tedious to add it to every page requiring consent for specific scripts.

JavaScript:
```javascript
onMounted(() => {
  if (window && window.Cookiebot) {
    window.Cookiebot.runScripts();
  }
});
```
TypeScript:
```ts
onMounted(() => {
  if (window && window.Cookiebot) {
    (window.Cookiebot as any).runScripts();
  }
});
```

**Comment by bnachtweh** [comment]
Thank you for your issue @wavedeck. I will look into your issue and will get back to you after reproduction

**Comment by bnachtweh** [comment]
@wavedeck your issue should have been addressed in the new 3.0.1 release. Please let me know if it works as expected :-)

**Comment by wavedeck** [comment]
@bnachtweh with my use-case (manual blocking mode) everything works as expected. I didn’t however test the automatic script blocking mode but i guess it should work for it too
